### ISSUE TITLE
Error when opening transfers page

### ISSUE THREAD

**Issue by jonashellmann** [akaunting/akaunting#1374]
Just by opening the transfers page (/banking/transfers), I get an error message like the one shown below. I think the problem started after upgrading to version 2.0.6 of Akaunting.

### System information (Akaunting, PHP versions)

* PHP version: 7.4.4
* Akaunting version: 2.0.7

### Error Message
<pre>
ErrorException:
Trying to get property 'account' of non-object

  at /var/www/html/akaunting/app/Http/Controllers/Banking/Transfers.php:36
  at Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
     (/var/www/html/akaunting/app/Http/Controllers/Banking/Transfers.php:36)
  at App\Http\Controllers\Banking\Transfers->index()
  at call_user_func_array()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Controller.php:54)
  at Illuminate\Routing\Controller->callAction()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php:45)
  at Illuminate\Routing\ControllerDispatcher->dispatch()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Route.php:225)
  at Illuminate\Routing\Route->runController()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Route.php:182)
  at Illuminate\Routing\Route->run()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Router.php:681)
  at Illuminate\Routing\Router->Illuminate\Routing\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:128)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/santigarcor/laratrust/src/Middleware/LaratrustPermission.php:32)
  at Laratrust\Middleware\LaratrustPermission->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/Money.php:54)
  at App\Http\Middleware\Money->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/DateFormat.php:35)
  at App\Http\Middleware\DateFormat->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/santigarcor/laratrust/src/Middleware/LaratrustPermission.php:32)
  at Laratrust\Middleware\LaratrustPermission->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/AdminMenu.php:30)
  at App\Http\Middleware\AdminMenu->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/RedirectIfWizardNotCompleted.php:21)
  at App\Http\Middleware\RedirectIfWizardNotCompleted->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/LogoutIfUserDisabled.php:21)
  at App\Http\Middleware\LogoutIfUserDisabled->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:29)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:29)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/firewall/src/Abstracts/Middleware.php:36)
  at Akaunting\Firewall\Abstracts\Middleware->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/akaunting/language/src/Middleware/SetLocale.php:95)
  at Akaunting\Language\Middleware\SetLocale->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/LoadCurrencies.php:27)
  at App\Http\Middleware\LoadCurrencies->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/LoadSettings.php:27)
  at App\Http\Middleware\LoadSettings->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/AddXHeader.php:18)
  at App\Http\Middleware\AddXHeader->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/app/Http/Middleware/RedirectIfNotInstalled.php:21)
  at App\Http\Middleware\RedirectIfNotInstalled->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php:41)
  at Illuminate\Routing\Middleware\SubstituteBindings->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Auth/Middleware/Authenticate.php:44)
  at Illuminate\Auth\Middleware\Authenticate->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php:76)
  at Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php:49)
  at Illuminate\View\Middleware\ShareErrorsFromSession->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:56)
  at Illuminate\Session\Middleware\StartSession->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php:37)
  at Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php:66)
  at Illuminate\Cookie\Middleware\EncryptCookies->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103)
  at Illuminate\Pipeline\Pipeline->then()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Router.php:683)
  at Illuminate\Routing\Router->runRouteWithinStack()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Router.php:658)
  at Illuminate\Routing\Router->runRoute()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Router.php:624)
  at Illuminate\Routing\Router->dispatchToRoute()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Routing/Router.php:613)
  at Illuminate\Routing\Router->dispatch()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:165)
  at Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:128)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php:65)
  at Barryvdh\Debugbar\Middleware\InjectDebugbar->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21)
  at Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21)
  at Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php:27)
  at Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/misterphilip/maintenance-mode/src/MisterPhilip/MaintenanceMode/Http/Middleware/CheckForMaintenanceMode.php:108)
  at MisterPhilip\MaintenanceMode\Http\Middleware\CheckForMaintenanceMode->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/fideloper/proxy/src/TrustProxies.php:57)
  at Fideloper\Proxy\TrustProxies->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/dingo/api/src/Http/Middleware/Request.php:111)
  at Dingo\Api\Http\Middleware\Request->handle()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:167)
  at Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103)
  at Illuminate\Pipeline\Pipeline->then()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:140)
  at Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
     (/var/www/html/akaunting/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:109)
  at Illuminate\Foundation\Http\Kernel->handle()
     (/var/www/html/akaunting/index.php:27)
</pre>

**Comment by denisdulici** [comment]
I just tried but wasn't able to reproduce this error. It seems your transactions have been deleted somehow. Feel free to use the forum for further support.

**Comment by AtjonTV** [comment]
I have the exact same issue .. @jonashellmann where you able to find a solution?

I upgraded from 1.3.17 up to 2.0.22 today.

**Comment by jonashellmann** [comment]
@AtjonTV I remember that there were corrupted entries in the database for transfers after I performed the upgrade. And after some try-and-error I was able to identify and delete/change them.
But I am not sure anymore how I exactly I found these corrupted records or why they appeared in the first place. Maybe there were bigger changes in the database schema.

**Comment by AtjonTV** [comment]
I checked and I found some entries that referenced a no-longer existing Account 2, I removed those but it would still error. I then removed all with Account 1 as that one is also not in use (I only use Bank Accounts 3 and 4). Still it wouldnt work.

I now simply cloned the table and truncated it. I may re add it after looking into it a bit more to find out whats wrong.

I was able to resolve all relations with success, so not sure whats broken with the old data :(
